### PR TITLE
Use HIO_HANDLE for depacker input

### PR DESCRIPTION
**Pull request by ccawley2011**

### CODE REVIEW

**Review by AliceLR — APPROVED** [review]
I think this should wait for #495 to see if that fixes the coverage drops, but otherwise this looks good to me.

**Comment by sezero** [comment]
#495 is in. Should we merge this?

**Comment by AliceLR** [comment]
> #495 is in. Should we merge this?

I'll rebase this to master to make sure Codecov is happy first, otherwise, I think this is OK. It might also be worth getting @cmatsuoka's feedback on this if he's available.

File: src/depackers/depacker.c [hunk 1]
```diff
@@ -205,16 +205,15 @@ int libxmp_decrunch(HIO_HANDLE **h, const char *filename, char **temp)
...
 	int headersize;
 	int i;
 	struct depacker *depacker = NULL;
 
 	cmd[0] = NULL;
 	*temp = NULL;
-	f = (*h)->handle.file;
 
-	headersize = fread(b, 1, 1024, f);
+	headersize = hio_read(b, 1, 1024, *h);
```
[the remark this answers is not in the document]
Comment (AliceLR) [reply src/depackers/depacker.c:216]: The Codecov comment and report on this pull request are pretty useless, but I found the rebased commit in the commits list and it *looks* like that patch fixed this.

**Comment by sezero** [comment]
> It might also be worth getting @cmatsuoka's feedback on this if he's available.

OK.

**Comment by sezero** [comment]
PING?

**Comment by sezero** [comment]
Rebase? PING?

**Comment by sezero** [comment]
PING? What do we do?

**Comment by sezero** [comment]
I'm merging this one: all checks are passing, and the patch is sane.

**Comment by AliceLR** [comment]
My main concern here was that this *might* conflict with Claudio's musings about separating the depackers out, but there's no reason in the event of that that the HIO stuff couldn't just be duplicated.

Since Claudio can't provide feedback currently, and working around these changes is trivial if-needed, I think this is fine. This (and #496 and #498) are relatively uncontroversial but this pattern will be a HUGE problem going forward when we need to e.g. change the API. We can at best only *guess* whether or not we're stepping on his toes in the current situation. It is frustrating and is part of why I've been silent lately. I'll finish up verifying and merging #496 and #498 soon, sorry.

**Pull request merged**